tests/samples_smoke_tests/common/samples_common_test_clas.py

```diff
@@ -0,0 +1,432 @@
+            elif ('m_hp' == k):
+                param['m_hp'] = SamplesCommonTestClass.reset_models_path(param['m_hp'])
+            elif ('m_va' == k):
+                param['m_va'] = SamplesCommonTestClass.reset_models_path(param['m_va'])
+            elif ('m_lpr' == k):
+                param['m_lpr'] = SamplesCommonTestClass.reset_models_path(param['m_lpr'])
+            elif ('m_em' == k):
+                param['m_em'] = SamplesCommonTestClass.reset_models_path(param['m_em'])
+            elif ('m_pa' == k):
+                param['m_pa'] = SamplesCommonTestClass.reset_models_path(param['m_pa'])
+            elif ('m_reid' == k):
+                param['m_reid'] = SamplesCommonTestClass.reset_models_path(param['m_reid'])
+            elif ('m_fd' == k):
+                param['m_fd'] = SamplesCommonTestClass.reset_models_path(param['m_fd'])
+            elif ('m_act' == k):
+                param['m_act'] = SamplesCommonTestClass.reset_models_path(param['m_act'])
+            elif ('m_lm' == k):
+                param['m_lm'] = SamplesCommonTestClass.reset_models_path(param['m_lm'])
+            elif ('m_det' == k):
+                param['m_det'] = SamplesCommonTestClass.reset_models_path(param['m_det'])
+            elif ('m_td' == k):
+                param['m_td'] = SamplesCommonTestClass.reset_models_path(param['m_td'])
+            elif ('m_tr' == k):
+                param['m_tr'] = SamplesCommonTestClass.reset_models_path(param['m_tr'])
+            elif ('m_en' == k):
+                param['m_en'] = SamplesCommonTestClass.reset_models_path(param['m_en'])
+            elif ('m_de' == k):
+                param['m_de'] = SamplesCommonTestClass.reset_models_path(param['m_de'])
+            elif ('l' == k and 'pascal_voc_classes' in param['l']):
+                param['l'] = os.path.join(Environment.env['test_data'], param['l'])
+            elif ('pp' == k):
+                param['pp'] = gpu_lib_path
+            elif ('r' == k):
+                if len(param['r']) > 0:
+                    param['r'] = os.path.join(Environment.env['test_data'], param['r'])
+            elif ('o' == k):
+                param['o'] = os.path.join(Environment.env['out_directory'], param['o'])
+            elif ('wg' == k):
+                param['wg'] = os.path.join(Environment.env['out_directory'], param['wg'])
+            elif ('we' == k):
+                param['we'] = os.path.join(Environment.env['out_directory'], param['we'])
+            elif ('fg' == k):
+                param['fg'] = os.path.join(Environment.env['test_data'], param['fg'])
+            elif ('labels' == k):
+                label_folder = os.path.dirname(executable_path.split(' ')[-1])
+                param['labels'] = os.path.join(label_folder, param['labels'])
+            elif ('lb') == k:
+                param['lb'] = os.path.join(Environment.env['test_data'], param['lb'])
+
+    @staticmethod
+    def get_cmd_line(param, use_preffix=True, long_hyphen=[]):
```

Comment:
it's not accurate to use mutable types in function definition - https://towardsdatascience.com/python-pitfall-mutable-default-arguments-9385e8265422